illumos-utils/src/zfs.rs

```diff
@@ -796,6 +799,21 @@ fn is_directory_immutable(
     return Ok(result);
 }
 
+struct DatasetMountInfo {
```

Comment:
We used to return a tuple of bools, I'm just making this explicit for readability.